lib/luanetfilter.c

```diff
@@ -89,19 +191,27 @@ static const lunatik_class_t luanetfilter_class = {
 	.sleep = false,
 };
 
-static int luanetfilter_xtable(lua_State *L)
+static void luanetfilter_release(void *private)
 {
-	lunatik_object_t *object;
-	luanetfilter_xtable_t *luaxtable;
-
-	object = lunatik_newobject(L, &luanetfilter_class, sizeof(luanetfilter_xtable_t));
-	luaxtable = (luanetfilter_xtable_t *)object->private;
+	if (match) {
+		xt_unregister_match(match);
+		lunatik_free(match);
+		match = NULL;
```

Comment:
you can move this to a "template" as well.. `luanetfilter_xtable_unreg`

Reply:
This will not be required in the future versions when we introduce rcu hashtable, at that time we get the object from the registry and free it.

Reply:
I don't think it should be the case here.. you have the object and it should hold references to match and target structures.. btw, I would do this change promptly.. that is, move `match` and `target` structures to `luanetfilter_xtable_t`.. and having a global `luanetfilter` object while you don't have the RCU table

Reply:
> Can I proceed like this ?

yup.. I think it's a good approach, I would just get rid of the bitfield.. it doesn't make much sense here, IMHO.. as it's not a packed structure, we will have padding there anyway.. also, I would have typedef for this enum and name it `luanetfilter_type_t` to simplify the namespace..

I really like this.. so we can have a the same `:unregister()` method for any kind of hook.. good one ;-)

Reply:
Yeah, I thought of having `unregister()` method for these, that's why I came up with this